ajax/displaypreference.php

```diff
@@ -53,9 +55,9 @@
     }
 } else if (isset($_POST['action']) && $_POST['action'] === 'update_order') {
     if (!isset($_POST['itemtype'], $_POST['users_id'], $_POST['opts'])) {
-        die(400);
+        throw new BadRequestHttpException('Missing input keys');
```

Comment:
No need to add a specific message to these exceptions.

```suggestion
        throw new BadRequestHttpException();
```